projects/dsp-ui/src/lib/search/expert-search/expert-search.component.spec.ts

```diff
@@ -0,0 +1,134 @@
+
+    ngOnInit() {
+    }
+
+}
+
+describe('ExpertSearchComponent', () => {
+  let testHostComponent: TestHostComponent;
+  let testHostFixture: ComponentFixture<TestHostComponent>;
+  let hostCompDe: DebugElement;
+
+  const dspConfSpy = new KnoraApiConfig('http', 'localhost', 3333, undefined, undefined, true);
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+        declarations: [
+            ExpertSearchComponent,
+            TestHostComponent
+        ],
+        imports: [
+            FormsModule,
+            ReactiveFormsModule,
+            RouterTestingModule,
+            BrowserAnimationsModule,
+            MatFormFieldModule,
+            MatInputModule
+        ],
+        providers: [
+            {
+                provide: DspApiConfigToken,
+                useValue: dspConfSpy
+            },
+            {
+                provide: ActivatedRoute,
+                useValue: {
+                params: null
+                }
+            }
+        ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    testHostFixture = TestBed.createComponent(TestHostComponent);
+    testHostComponent = testHostFixture.componentInstance;
+
+    testHostFixture.detectChanges();
+
+    hostCompDe = testHostFixture.debugElement;
+  });
+
+  it('should create', () => {
+    expect(testHostComponent).toBeTruthy();
+    expect(testHostComponent.expertSearch).toBeTruthy();
+  });
+
+  it('should init the form with the default query', () => {
+    const textarea = hostCompDe.query(By.css('textarea.textarea-field-content'));
+    const textareaEle = textarea.nativeElement;
+
+    expect(textareaEle.value).toBe(
+        `PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+PREFIX incunabula: <http://localhost:3333/ontology/0803/incunabula/simple/v2#>
+
+CONSTRUCT {
+    ?book knora-api:isMainResource true .
+    ?book incunabula:title ?title .
+
+} WHERE {
+    ?book a incunabula:book .
+    ?book incunabula:title ?title .
+}
+`
+    );
+  });
+
+  it('should reset the form', () => {
+
+    const resetBtn = hostCompDe.query(By.css('button.reset'));
+    const textarea = hostCompDe.query(By.css('textarea.textarea-field-content'));
+
+    const resetEle = resetBtn.nativeElement;
+    const textareaEle = textarea.nativeElement;
+
+    // delete textarea content displayed by default to make a change
+    textareaEle.value = '';
+    expect(textareaEle.value).toBe('');
+
+    resetEle.click();
+
+    testHostFixture.detectChanges();
+
+    // reset the textarea content
+    expect(textareaEle.value).toBe(
+        `PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+PREFIX incunabula: <http://localhost:3333/ontology/0803/incunabula/simple/v2#>
+
+CONSTRUCT {
+    ?book knora-api:isMainResource true .
+    ?book incunabula:title ?title .
+
+} WHERE {
+    ?book a incunabula:book .
+    ?book incunabula:title ?title .
+}
+`
+    );
+  });
+
```

Comment:
you should add an  `it` that checks for the correct registration of the query in the params service.

I think this is identical to what I did for the Gravsearch generation service:

https://github.com/dasch-swiss/dsp-ui-lib/blob/b3b1fee1ecc97085f3bdb199c0352c36825b206a/projects/dsp-ui/src/lib/search/services/gravsearch-generation.service.spec.ts#L14-L31

https://github.com/dasch-swiss/dsp-ui-lib/blob/b3b1fee1ecc97085f3bdb199c0352c36825b206a/projects/dsp-ui/src/lib/search/services/gravsearch-generation.service.spec.ts#L78-L80

Reply:
done in 062ccea

Reply:
ok